geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/RsuIndicators.groovy

```diff
@@ -71,6 +71,113 @@ IProcess freeExternalFacadeDensity() {
+ * Reference Spatial Unit (RSU) divided by the RSU area. The calculation is performed more accurately than
+ * in the freeExternalFacadeDensity IProcess since in this process only the part of building being in the RSU
+ * is considered in the calculation of the indicator.
+ *
+ * @param datasource A connexion to a database (H2GIS, PostGIS, ...) where are stored the input Table and in which
+ * the resulting database will be stored
+ * @param buildingTable The name of the input ITable where are stored the buildings geometry, the building height,
+ * the building and the rsu id
+ * @param rsuTable The name of the input ITable where are stored the rsu geometries and the id_rsu
+ * @param prefixName String use as prefix to name the output table
+ *
+ * @return A database table name.
+ * @author Jérémy Bernard
+ */
+IProcess freeExternalFacadeDensityExact() {
+    return create {
+        title "RSU free external facade density (exact version)"
+        id "freeExternalFacadeDensityExact"
+        inputs buildingTable: String, rsuTable: String, prefixName: String, datasource: JdbcDataSource
+        outputs outputTableName: String
+        run { buildingTable, rsuTable, prefixName, datasource ->
+
+            def GEOMETRIC_FIELD_RSU = "the_geom"
+            def GEOMETRIC_FIELD_BU = "the_geom"
+            def ID_FIELD_RSU = "id_rsu"
+            def ID_FIELD_BU = "id_build"
+            def HEIGHT_WALL = "height_wall"
+            def FACADE_AREA = "facade_area"
+            def RSU_AREA = "rsu_area"
+            def BASE_NAME = "_exact_free_external_facade_density"
+
+            debug "Executing RSU free external facade density (exact version)"
+
+            // The name of the outputTableName is constructed
+            def outputTableName = prefix prefixName, BASE_NAME
+
+            // Temporary table names
+            def buildLine = postfix "buildLine"
+            def buildLineRsu = postfix "buildLineRsu"
+            def sharedLineRsu = postfix "shareLineRsu"
+
+            // Consider facades as touching each other within a snap tolerance
+            def snap_tolerance = 0.01
+
+            // 1. Convert the building polygons into lines and create the intersection with RSU polygons
+            datasource."$buildingTable"."$GEOMETRIC_FIELD_RSU".createSpatialIndex()
+            datasource."$rsuTable"."$GEOMETRIC_FIELD_RSU".createSpatialIndex()
+            datasource """
+                DROP TABLE IF EXISTS $buildLine;
+                CREATE TABLE $buildLine
+                    AS SELECT   a.$ID_FIELD_BU, a.$ID_FIELD_RSU, ST_AREA(b.$GEOMETRIC_FIELD_RSU) AS $RSU_AREA,
+                                ST_INTERSECTION(ST_TOMULTILINE(a.$GEOMETRIC_FIELD_BU), b.$GEOMETRIC_FIELD_RSU) AS $GEOMETRIC_FIELD_BU,
+                                a.$HEIGHT_WALL
+                    FROM $buildingTable AS a LEFT JOIN $rsuTable AS b
+                    ON a.$ID_FIELD_RSU = b.$ID_FIELD_RSU""".toString()
```

Comment:
datasource."$buildingTable"."$ID_FIELD_RSU".createIndex()
  datasource."$rsuTable"."$ID_FIELD_RSU".createIndex()

Reply:
OK I have replaced the index creation accordingly